pkg/virt-api/validating-webhook/validating-webhook.go

```diff
@@ -321,6 +321,25 @@ func validateVirtualMachineSpec(field *k8sfield.Path, spec *v1.VirtualMachineSpe
 		return causes
 	}
 
+	// Validate memory size if values are not negative
+	if spec.Domain.Resources.Requests.Memory().Value() < 0 {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("%s is specified as negative value", field.Child("domain", "resources", "requests", "memory").String()),
+			Field:   field.Child("domain", "resources", "requests", "memory").String(),
+		})
+		// We won't process anything with negative requests.memory size
+		return causes
+	} else if spec.Domain.Resources.Limits.Memory().Value() < 0 {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("%s is specified as negative value", field.Child("domain", "resources", "limits", "memory").String()),
+			Field:   field.Child("domain", "resources", "limits", "memory").String(),
+		})
+		// We won't process anything with negative limits.memory size
+		return causes
```

Comment:
Don't return early. We want to go through the whole struct and collect all issues we can find. We can then report all issues at once to the user.

Reply:
Done